File: antigravity/scratch/automind-ai/frontend/types/ai.ts
import { Car } from "./car";

export type ChatRole = "user" | "assistant" | "system";

export interface ChatMessage {
  id: string;
  role: ChatRole;
  content: string;
  timestamp: string;
  suggestedActions?: string[];
  referencedCarIds?: string[];
}

export interface AssistantChatRequest {
  sessionId?: string;
  message: string;
  contextCarId?: string;
  preferences?: {
    budgetMax?: number;
    usage?: "daily" | "family" | "track" | "offroad" | "luxury";
  };
}

export interface AssistantChatResponse {
  sessionId: string;
  reply: string;
  suggestedQuestions: string[];
  recommendedCars?: Car[];
}

export interface RecommendationRequest {
  budgetMin: number;
  budgetMax: number;
  primaryUse: "commute" | "family" | "track" | "roadtrip" | "utility";
  preferredFuelTypes: string[];
  mustHaveFeatures: string[];
  aestheticPreference: "minimalist" | "aggressive" | "luxury" | "futuristic";
}

export interface RecommendationMatch {
  car: Car;
  matchScore: number; // 0 - 100
  aiRationale: string;
  pros: string[];
  cons: string[];
}

export interface RecommendationResponse {
  summary: string;
  topMatches: RecommendationMatch[];
}

File: antigravity/scratch/automind-ai/backend/app/schemas/ai.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, Field
from app.schemas.car import CarResponse


class AssistantChatRequest(BaseModel):
    session_id: Optional[str] = Field(None, alias="sessionId")
    message: str
    context_car_id: Optional[str] = Field(None, alias="contextCarId")
    preferences: Optional[Dict[str, Any]] = None

    class Config:
        populate_by_name = True


class AssistantChatResponse(BaseModel):
    session_id: str = Field(alias="sessionId")
    reply: str
    suggested_questions: List[str] = Field(alias="suggestedQuestions")
    recommended_cars: Optional[List[CarResponse]] = Field(None, alias="recommendedCars")

    class Config:
        populate_by_name = True


class RecommendationRequest(BaseModel):
    budget_min: float = Field(alias="budgetMin")
    budget_max: float = Field(alias="budgetMax")
    primary_use: str = Field(alias="primaryUse")
    preferred_fuel_types: List[str] = Field(alias="preferredFuelTypes")
    must_have_features: List[str] = Field(alias="mustHaveFeatures")
    aesthetic_preference: str = Field(alias="aestheticPreference")

    class Config:
        populate_by_name = True


class RecommendationMatch(BaseModel):
    car: CarResponse
    match_score: int = Field(alias="matchScore")
    ai_rationale: str = Field(alias="aiRationale")
    pros: List[str]
    cons: List[str]

    class Config:
        populate_by_name = True


class RecommendationResponse(BaseModel):
    summary: str
    top_matches: List[RecommendationMatch] = Field(alias="topMatches")

    class Config:
        populate_by_name = True
